tedana/io.py

```diff
@@ -225,6 +229,14 @@ def save_img(self, data, name):
                 "Data must have number of dimensions in (1, 2), not "
                 f"{data.ndim}"
             )
+        # Coerce data to be 32-bit max in the cases of float64, int64
+        # Note that int64 niftis cannot be read by mricroGL, AFNI
+        vox_type = data.dtype
+        if vox_type == np.int64:
```

Comment:
What about `np.issubdtype(data.dtype, np.integer)`? That should be more robust to other int types, like `int8`.

Reply:
I don't want to upscale an int8 to an int32 if we don't have to, as it just wastes space (and time, since the upcast requires allocating a large 32-bit integer array). The main issue is storing something not widely usable (int64) or something that has more precision than needed (float64), so I singled those two types out.

Reply:
That makes sense. Is int64 the only datatype not usable by AFNI and/or mricroGL?

Reply:
I'm guessing that `uint64` is probably not supported if `int64` is not, but there's no reason to suppose that we'll end up with `uint64` arrays unless we explicitly ask for them, which seems unlikely and solvable when we get to it if so.

Reply:
Thanks for clarifying. The way you have sounds good then.